src/cli/cli_history.cpp

```diff
@@ -1554,6 +1565,126 @@ const char *History::DnsSrpAddrTypeToString(otHistoryTrackerDnsSrpAddrType aType
     return Stringify(aType, kAddrTypeStrings, "--");
 }
 
+#if OPENTHREAD_CONFIG_HISTORY_TRACKER_CLIENT_ENABLE
+
+otError History::ParseQueryArgs(Arg       aArgs[],
+                                bool     &aIsList,
+                                uint16_t &aRloc16,
+                                uint16_t &aNumEntries,
+                                uint32_t &aMaxEntryAge) const
+{
+    otError error = OT_ERROR_NONE;
+
+    if (*aArgs == "list")
+    {
+        aArgs++;
+        aIsList = true;
+    }
+    else
+    {
+        aIsList = false;
+    }
+
+    SuccessOrExit(error = aArgs->ParseAsUint16(aRloc16));
+    aArgs++;
+
+    if (aArgs->ParseAsUint16(aNumEntries) == OT_ERROR_NONE)
+    {
+        aArgs++;
+    }
+    else
+    {
+        aNumEntries = 0;
+    }
+
+    if (aArgs->ParseAsUint32(aMaxEntryAge) == OT_ERROR_NONE)
+    {
+        aArgs++;
+    }
+    else
+    {
+        aMaxEntryAge = 0;
+    }
+
+    error = aArgs[0].IsEmpty() ? OT_ERROR_NONE : OT_ERROR_INVALID_ARGS;
+
+exit:
+    return error;
+}
+
+template <> otError History::Process<Cmd("query")>(Arg aArgs[])
+{
+    otError  error = OT_ERROR_NONE;
+    uint16_t rloc16;
+    uint16_t maxEntries;
+    uint32_t maxEntryAge;
+
+    /**
+     * @cli history query netinfo
+     * @code
+     * history query netinfo 0xac00
+     * | Age                  | Role     | Mode | RLOC16 | Partition ID |
+     * +----------------------+----------+------+--------+--------------+
+     * |         00:00:36.786 | detached | rdn  | 0xac00 |    807291876 |
+     * |         00:00:43.966 | detached | rdn  | 0xfffe |            0 |
+     * Done
+     * @endcode
+     * @cparam history query netinfo [@ca{list}] @ca{rloc16} [@ca{num-entries}] [@ca{max-entry-age}]
```

Comment:
Should the rloc16 parameter be listed first?  Since it's not optional

Reply:
I prefer to keep the current arg structure where the `list` keyword is optional but, if present, must be placed immediately after `netinfo` and before the `rloc16`.

My intention is to make the syntax similar to the existing `history` command. This new `query` command should feel like an extension of the old one; instead of getting history information from the local device, we are requesting it from a remote device specified by its `rloc16`.